course_catalog/etl/see.py

```diff
@@ -0,0 +1,274 @@
+    runs = []
+    for date_string in date_strings:
+        match = re.match(pattern_1_month, date_string)
+        if match:
+            start_date = datetime.strptime(
+                f"{match.group('start_m')} {match.group('start_d')} {match.group('year')}",
+                "%b %d %Y",
+            )
+            end_date = datetime.strptime(
+                f"{match.group('start_m')} {match.group('end_d')} {match.group('year')}",
+                "%b %d %Y",
+            )
+            runs.append(
+                (start_date.replace(tzinfo=pytz.utc), end_date.replace(tzinfo=pytz.utc))
+            )
+            continue
+        match = re.match(pattern_1_year, date_string)
+        if match:
+            start_date = datetime.strptime(
+                f"{match.group('start_m')} {match.group('start_d')} {match.group('year')}",
+                "%b %d %Y",
+            )
+            end_date = datetime.strptime(
+                f"{match.group('end_m')} {match.group('end_d')} {match.group('year')}",
+                "%b %d %Y",
+            )
+            runs.append(
+                (start_date.replace(tzinfo=pytz.utc), end_date.replace(tzinfo=pytz.utc))
+            )
+            continue
+        match = re.match(pattern_2_years, date_string)
+        if match:
+            start_date = datetime.strptime(
+                f"{match.group('start_m')} {match.group('start_d')} {match.group('start_y')}",
+                "%b %d %Y",
+            )
+            end_date = datetime.strptime(
+                f"{match.group('end_m')} {match.group('end_d')} {match.group('end_y')}",
+                "%b %d %Y",
+            )
+            runs.append(
+                (start_date.replace(tzinfo=pytz.utc), end_date.replace(tzinfo=pytz.utc))
+            )
+    return runs
+
+
+def _parse_instructors(details):
+    """
+    Extract instructor names from the course detail page
+
+    Args:
+        details(Tag): BeautifulSoup Tag for course details
+
+    Returns:
+        list of dict: List of first & last names of each instructor
+
+    """
+    instructors = [
+        instructor.find("h2").get_text().split()
+        for instructor in details.findAll("article", {"class": "faculty-article"})
+    ]
+    return [
+        {"first_name": instructor[0], "last_name": instructor[1]}
+        for instructor in instructors
+    ]
+
+
+def _parse_short_description(details):
+    """
+    Extract short description from the course detail page
+
+    Args:
+        details(Tag): BeautifulSoup Tag for course details
+
+    Returns:
+        str: Short course description
+
+    """
+    section = details.find("section", {"class": "lead-content block-inner"})
+    # pylint: disable=expression-not-assigned
+    [header.extract() for header in section.findAll("header")]
+    # pylint: disable=expression-not-assigned
+    [style.extract() for style in section.findAll("style")]
+    return strip_extra_whitespace(section.get_text(separator=" ").strip())
```

Comment:
Are we mutating the input argument `details` by doing this?

Reply:
Yes, It is modifying this section of it, which is never used again.